source/_posts/2018-05-30-home-assistant-security.markdown

```diff
@@ -0,0 +1,250 @@
+|   challenge_response: supported
+|_  message_signing: disabled (dangerous, but default)
+| smb2-security-mode: 
+|   2.02: 
+|_    Message signing enabled but not required
+| smb2-time: 
+|   date: 2018-05-29 16:41:05
+|_  start_date: 1601-01-01 00:53:28
+
+TRACEROUTE
+HOP RTT     ADDRESS
+1   0.46 ms 192.168.0.215
+
+OS and Service detection performed. Please report any incorrect results at https://nmap.org/submit/ .
+Nmap done: 1 IP address (1 host up) scanned in 727.43 seconds
+```
+
+139 and 445 are open and it's possible to enumerate the shares. With different tools you will get pretty much the same information.
+
+```bash
+$ smbclient -L //192.168.0.215 -U%
+
+	Sharename       Type      Comment
+	---------       ----      -------
+	config          Disk      
+	addons          Disk      
+	share           Disk      
+	backup          Disk      
+	IPC$            IPC       
+IPC Service (Samba HomeAssistant config share)
+Reconnecting with SMB1 for workgroup listing.
+
+	Server               Comment
+	---------            -------
+
+	Workgroup            Master
+	---------            -------
+	WORKGROUP            HASSIO
+```
+
+But without username and password you can't get access to the configuration file with the settings shown here.
+
+<p class='img'>
+  <img
+    src='/images/blog/2018-05-security/shares.png'
+    alt='Accessing Hass.io shares with GNOME' />
+  Accessing SMB shares on Hass.io
+</p>
+
+## {% linkable_title Allowing remote access %}
+
+Home Assistant is NOT able to change the configuration of your router or firewall. This means that you need to setup port-forwarding and adjust firewall rules if you want to allow access from the internet. By default your frontend, Mosquitto, SSH and your Samba shares are only accessible from your local network.
+
+Beside the recommendation in the [Securing checklist](/docs/configuration/securing/) for remote access also check those points:
+
+- Don't forward the ports used by the Samba add-on.
+- Use `ip_ban_enabled` and `login_attempts_threshold`. A brute force attack on the frontend or SSH is a threat.
+- Disable `trusted_networks`.
+- Limit the access to the Home Assistant frontend. Don't allow access to the [Configurator add-on](/addons/configurator/) for the internet.
+- Check the [UPnP](https://en.wikipedia.org/wiki/Universal_Plug_and_Play) settings of your router/firewall and disable it when not needed.
+- Review your router/firewall logs on a regular basis.
+- If possible, use the VPN feature of your router or firewall to get access to your home network and Home Assistant. 
+
+## {% linkable_title Summary %}
+
+As you can see in the results of the scans, only the ports which are needed by a running services are open. If you stop an add-on then the port is no longer accessible and the porosity of your Hass.io machine will decrease.
+
+We don't have an unique server banner but in combination with the HTML title `Home Assistant`, is it simple to identify Home Assistant instances.
```

Comment:
"have an unique server banner" -> "have a unique server banner"
"is it simple" -> "it is simple"